### DISCUSSION TITLE
en/lcci/1.5/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#3907] [Announcements]
# en/lcci/1.5/

LeetCode solutions in any programming language

https://leetcode.doocs.org/en/lcci/1.5/

**Comment by rajputanisha** [comment]
#include <bits/stdc++.h>
using namespace std;

bool fun(string s1, string s2) {
    int len1 = s1.size();
    int len2 = s2.size();
    if (abs(len1 - len2) > 1) return false;

    int i = 0, j = 0;
    bool found = false;
    while (i < len1 && j < len2) {
        if (s1[i] != s2[j]) {
            if (found) return false;
            found = true;

       
            if (len1 > len2) {
                i++;
            } else if (len1 < len2) {
                j++;
            } else {
              i++;
                j++;
            }
        } else {
         i++;
           j++;
        }
    }

    return true;
}

int main() {
    string s1 = "pale";
    string s2 = "pal";
    cout << (fun(s1, s2) ? "True" : "False") << endl;

    return 0;
}
